libraries/hatch-server-user-management/src/AWSCognitoClient.ts

```diff
@@ -224,4 +243,40 @@ export default class AWSCognitoClient implements UserServiceClient {
     return pemCerts;
   }
   
+  private convertAWSErrorToUserManagementError(awsError: AWSError) {
+    switch (awsError.code) {
+      case 'CodeMismatchException' || 'ExpiredCodeException': {
+        return new UserManagementError(UserManagementErrorCodes.INVALID_CODE, awsError.message);
+      }
+      case 'InvalidPasswordException': {
+        return new UserManagementError(UserManagementErrorCodes.INVALID_PASSWORD, awsError.message);
+      }
+      case 'NotAuthorizedException': {
+        if (awsError.message.includes('Password attempts exceeded')) {
+          return new UserManagementError(UserManagementErrorCodes.ACCOUNT_LOCKED, awsError.message);
+        } else {
+          return new UserManagementError(UserManagementErrorCodes.UNAUTHORIZED, awsError.message);
+        }
+      }
+      case 'PasswordResetRequiredException': {
+        return new UserManagementError(UserManagementErrorCodes.ACCOUNT_LOCKED, awsError.message);
+      }
+      case 'TokenExpiredException': {
+        return new UserManagementError(UserManagementErrorCodes.EXPIRED_TOKEN, awsError.message);
+      }
+      case 'UsernameExistsException': {
+        return new UserManagementError(UserManagementErrorCodes.USERNAME_EXISTS, awsError.message);
+      }
+      case 'UserNotConfirmedException': {
+        return new UserManagementError(UserManagementErrorCodes.USER_NOT_CONFIRMED, awsError.message);
+      }
+      case 'UserNotFoundException': {
+        return new UserManagementError(UserManagementErrorCodes.USER_NOT_FOUND, awsError.message);
+      }
+      default: {
+        return new UserManagementError(UserManagementErrorCodes.INTERNAL_ERROR, awsError.message);
```

Comment:
In case it's not already in the message, prob a good idea to explicitly add awsError.code to the message.

Reply:
Have it locally, will push later

Reply:
updated